controller/rest/file_monitor.go

```diff
@@ -118,6 +132,10 @@ func replaceFedFileMonitorProfiles(profiles []*share.CLUSFileMonitorProfile, acc
 		clusHelper.DeleteFileAccessRuleTxn(txn, name.(string))
 	}
 
+	if hasError {
+		return false
+	}
+
 	if ok, err := txn.Apply(); err != nil || !ok {
 		log.WithFields(log.Fields{"ok": ok, "error": err}).Error("Atomic write to the cluster failed")
```

Comment:
Should we return false when the transaction fails to apply, like you fixed in another place?

Reply:
fixed.